### ISSUE TITLE
Port Kolibri flatpak to GNOME 44 runtime

### ISSUE THREAD

**Issue by dbnicholson** [endlessm/endless-key-flatpak#1]
See https://github.com/learningequality/kolibri-installer-gnome/issues/80.

**Comment by dylanmccall** [comment]
Alternatively, we can update to Gtk 4 at the same time: https://github.com/endlessm/kolibri-installer-gnome/pull/10.

If we'd prefer to just do the runtime update, you may run into a weird issue involving an incompatible version of python3-markdown. That can be resolved by cherry-picking this change: https://github.com/endlessm/kolibri-installer-gnome/commit/b9f7ac0cf68c5cb20d124577f98139a539f2a9c0.

**Comment by dbnicholson** [comment]
https://github.com/endlessm/kolibri-installer-gnome/pull/10 was merged, but now we need those changes in the upstream kolibri-installer-gnome. Richard was kind enough to merge another PR there while Dylan is out, so I think if we can make a PR of those changes and mark it as fixing https://github.com/learningequality/kolibri-installer-gnome/issues/80 we'll be all set. The Gtk4 PR already has the changes for using GNOME 44.

**Comment by GeorgesStavracas** [comment]
I was not aware of the upstream repo. I'll submit it there ASAP.

**Comment by dbnicholson** [comment]
This was merged into upstream kolibri-installer-gnome in https://github.com/learningequality/kolibri-installer-gnome/pull/81 and released as [2.3.0](https://github.com/learningequality/kolibri-installer-gnome/releases/tag/v2.3.0).

**Comment by dbnicholson** [comment]
The last part here is to fixup and land https://github.com/flathub/org.learningequality.Kolibri/pull/62 so the Kolibri Flathub app is actually on the GNOME 44 runtime. That's not a blocker for us to continue with Endless Key, but I think it would be nice to do.

**Comment by dbnicholson** [comment]
I'm going to move this over to @dylanmccall since he has all the privileges to finish it off. @GeorgesStavracas please move on to #7.

**Comment by dylanmccall** [comment]
Okay, this should all be completed, with a public release happening in flathub/org.learningequality.Kolibri#63. If someone would like to give it a quick last-minute test before Flathub automatically publishes the thing, you can run the following commands:

```
flatpak uninstall org.learningequality.Kolibri
flatpak install --system https://dl.flathub.org/build-repo/43766/org.learningequality.Kolibri.flatpakref
flatpak run org.learningequality.Kolibri
```

**Comment by dbnicholson** [comment]
I keep getting the following exception on startup when used with the system service:

```
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]: ERROR    2023-08-25 14:33:09,201 Internal Server Error: /en/kolibri_desktop_auth_plugin/login/1001:a52896c8a2b54e08b740522251f0c24b
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]: Traceback (most recent call last):
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     response = get_response(request)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     response = self.process_exception_by_middleware(e, request)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/views/generic/base.py", line 68, in view
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     return self.dispatch(request, *args, **kwargs)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/views/generic/base.py", line 88, in dispatch
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     return handler(request, *args, **kwargs)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri_desktop_auth_plugin/views.py", line 12, in get
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     user = authenticate(token=token)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/contrib/auth/__init__.py", line 70, in authenticate
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     user = _authenticate_with_backend(backend, backend_path, request, credentials)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri/dist/django/contrib/auth/__init__.py", line 116, in _authenticate_with_backend
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     return backend.authenticate(*args, **credentials)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri_desktop_auth_plugin/backends.py", line 38, in authenticate
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     user_details = self._get_user_details(token)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:   File "/app/lib/python3.10/site-packages/kolibri_desktop_auth_plugin/backends.py", line 21, in _get_user_details
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]:     proxy = Gio.DBusProxy.new_sync(
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]: gi.repository.GLib.GError: g-io-error-quark: Error calling StartServiceByName for org.learningequality.Kolibri.Daemon: Timeout was reached (24)
Aug 25 14:33:09 talkbox eos-kolibri-daemon[77482]: INFO     2023-08-25 14:33:09,312 127.0.0.1 - - "GET /en/kolibri_desktop_auth_plugin/login/1001:a52896c8a2b54e08b740522251f0c24b" 500 0 "" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.4 Safari/605.1.15"
```

**Comment by dylanmccall** [comment]
Oops! That was unfortunate. Sorry for releasing this without testing that. I created issue #10 to keep track of the need to have documented testing steps here, I reverted a bad change (and made a note to try again later), and I released a quick fix:

- https://github.com/learningequality/kolibri-installer-gnome/commit/e72fbf6ef8f0ca5542d0252bb37e1fc55e1ca9fc
- https://github.com/learningequality/kolibri-installer-gnome/issues/86
- https://github.com/flathub/org.learningequality.Kolibri/pull/64
